utils/zerovec/src/ule/mod.rs

```diff
@@ -233,7 +256,7 @@ pub unsafe trait VarULE: 'static {
     ///
     /// # Safety
     ///
-    /// In most cases, the implementation of this function should involve re-casting the pointer.
+    /// The implementation of this function should involve re-casting the pointer.
```

Comment:
Thought: We don't really need `as_byte_slice()` any more, either on ULE or VarULE, because the safety constraints of the trait mean that we can transmute.  At least, the function could be auto-implemented.  Same with from_byte_slice_unchecked.

Really, the only function we need is the parsing/validation function.

Reply:
So I thought about this and ultimately elected not to because I think for unsafe traits it's better to have the API being provided as functions as much as possible to make it easy to reason about. We could add a default implementation here though.

Reply:
Would you mind adding the default implementation?  (I think it's in scope for this PR.)